src/OpenTK.Mathematics/Geometry/Ray3.cs

```diff
@@ -0,0 +1,143 @@
+        [Pure]
+        public Vector3 GetPoint(float distance) {
+            return GetPoint(in this, distance);
+        }
+
+        /// <summary>
+        /// Get the closest point along the ray to a vector, with optional range clamping.
+        /// </summary>
+        /// <param name="ray">The ray</param>
+        /// <param name="vector">The vector</param>
+        /// <param name="nearClamp">Near clamp</param>
+        /// <param name="farClamp">Far clamp</param>
+        /// <returns>The closest point to the <paramref name="vector"/> along the <paramref name="ray"/>, within the clamped range.</returns>
+        [Pure]
+        public Vector3 GetClosestPoint(in Vector3 vector, float nearClamp = 0, float farClamp = float.PositiveInfinity) {
+            return GetClosestPoint(in this, in vector, nearClamp, farClamp);
+        }
+
+        /// <param name="ray">The ray.</param>
+        /// <param name="distance">The distance to travel along the ray.</param>
+        /// <returns>The point <paramref name="distance"/> units along the ray. </returns>
+        public static Vector3 GetPoint(in Ray3 ray, float distance) {
+            return ray.Origin + ray.Direction * distance;
+        }
+
+        /// <summary>
+        /// Get the closest point along the ray to a vector, with optional range clamping.
+        /// </summary>
+        /// <param name="ray">The ray</param>
+        /// <param name="vector">The vector</param>
+        /// <param name="nearClamp">Near clamp</param>
+        /// <param name="farClamp">Far clamp</param>
+        /// <returns>The closest point to the <paramref name="vector"/> along the <paramref name="ray"/>, within the clamped range.</returns>
+        public static Vector3 GetClosestPoint(in Ray3 ray, in Vector3 vector, float nearClamp = 0, float farClamp = float.PositiveInfinity) {
+            (nearClamp, farClamp) = (Math.Min(nearClamp, farClamp), Math.Max(nearClamp, farClamp));
+            float scalar = Vector3.Dot(vector - ray.Origin, ray.Direction);
+            scalar = Math.Clamp(scalar, nearClamp, farClamp);
+            return GetPoint(in ray, scalar);
+        }
+
+        /// <summary>
+        /// Checks if a ray overlaps a sphere.
+        /// </summary>
+        /// <param name="ray">The ray to test.</param>
+        /// <param name="sphere">The sphere to test.</param>
+        /// <param name="hit">The nearest hit location, if applicable.</param>
+        /// <param name="nearClamp">Distance to clamp the line on the near side. Defaults to 0 to avoid objects behind the origin from being included. Set to float.NegativeInfinity to simulate a regular line.</param>
+        /// <param name="farClamp">Distance to clamp the line to the far side. Sets the max length of the ray for the intersection query.</param>
+        /// <returns></returns>
+        public static bool Intersects(in Ray3 ray, in Sphere sphere, out Vector3 hit, float nearClamp = 0, float farClamp = float.PositiveInfinity) {
```

Comment:
I think maybe we want to make it explicit in the function name what the shape we are testing against. Maybe something like `IntersectsSphere`?
I'm not sure this is the way to go, but it's my gut feeling atm.